### PR TITLE
Cub/3

### PR DESCRIPTION
**Pull request by jberkema**

### CODE REVIEW

File: src/parser/map_init.c [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+{
+	int		rows;
+	int		max_length;
+
+	rows = 0;
+	max_length = 0;
+	while (tmp != NULL)
+	{
+		rows++;
+		max_length = set_max_len(tmp, max_length);
```
Comment (letsgogeeky) [inline src/parser/map_init.c:76]: that aligns perfectly with how the validator works! thank you!

File: src/parser/map_init.c [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	{
+		m->map[i][j] = tmp[j];
+		j++;
+	}
+	if (nl != 0)
+		max_length--;
+	while (j < max_length)
+		m->map[i][j++] = ' ';
+	if (nl != 0)
+		m->map[i][j] = '\n';
```
Comment (letsgogeeky) [inline src/parser/map_init.c:29]: do we need to write the `\n` to the row array?

**Review by letsgogeeky — APPROVED** [review]
Looks great! added minor comments.

File: src/parser/map_init.c (hunk 2 above)
Comment (jberkema) [reply src/parser/map_init.c:29]: "The map must be parsed as it looks in the file" is written in the subject. I don't know how to interpret this in regard of nl. The task is more into the direction of spaces, at least it seems like in the subject: "Spaces are a valid part of the map and are up to you to handle. You must be able to parse any kind of map, as long as it respects the rules of the map."

**Pull request merged**